File: frontend/src/app/models/Carpooling/trip-price-suggestion.ts
export interface TripPriceSuggestion {
  suggestedPrice: number;
  basePrice: number;
  minHistoricalPrice: number;
  maxHistoricalPrice: number;
  similarTripsCount: number;
  holidayAdjusted: boolean;
  holidayName?: string;
  message: string;
}

File: backend/src/main/java/org/example/backend_tunisiahub/Services/Carpooling/TripPriceSuggestion.java
package org.example.backend_tunisiahub.Services.Carpooling;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class TripPriceSuggestion {

    private BigDecimal suggestedPrice;

    private BigDecimal basePrice;

    private BigDecimal minHistoricalPrice;

    private BigDecimal maxHistoricalPrice;

    private Integer similarTripsCount;

    private Boolean holidayAdjusted;

    private String holidayName;

    private String message;
}
